### DISCUSSION TITLE
🔒 Loosen Runtime Bytecode Check

### DISCUSSION THREAD

**Discussion by QYuQianchen** [pcaversaccio/createx#39] [Ideas]
It can be an edge case, users may be interested in creating a contract without any runtime bytecode, such as deploying with `0x600060006000f0`. Could implementing a check like `newContract.code.length == 0` be considered overly strict in this context?

**Comment by pcaversaccio** [comment]
@QYuQianchen in the [Design Principles](https://github.com/pcaversaccio/createx?tab=readme-ov-file#design-principles) we write the following:

- [`CreateX`](https://github.com/pcaversaccio/createx/blob/main/src/CreateX.sol) should cover _most_ but not all contract creation use cases.

I consider deploying a zero-byte runtime bytecode in the context of `CreateX` rather a bug than a feature & a non-use-case, and thus we should prevent people from accidentally deploying such a bytecode. @mds1 thoughts?

**Reply by mds1** [reply]
@QYuQianchen what is the use case for that? Currently I agree with @pcaversaccio, if your deployment results in empty bytecode you likely made a mistake

**Reply by pcaversaccio** [reply]
Actually, if someone wants to do this, anyone can deploy via `CreateX` another factory contract that allows for a zero-byte runtime code that can be used within the same deployment transaction to deploy such a zero-byte contract. Also, I don't see any reason why a specific deployed address matters in that case. I will close this discussion accordingly.
